examples/stable-diffusion/training/README.md

```diff
@@ -163,86 +163,92 @@ Install the requirements:
 pip install -r requirements.txt
 ```
 
-### Example for SDXL
-We can launch the fine-tuning of SDXL model using:
+### Single-card Training
 
 ```bash
 python train_text_to_image_sdxl.py \
   --pretrained_model_name_or_path stabilityai/stable-diffusion-xl-base-1.0 \
   --pretrained_vae_model_name_or_path stabilityai/sdxl-vae \
   --dataset_name lambdalabs/pokemon-blip-captions \
-  --resolution 1024 \
+  --resolution 512 \
+  --crop_resolution 512 \
```

Comment:
If we use the same value, maybe `crop_resolution` is not needed. Or do we want to give users the possibility to specify different crop sizes?
I replied in the other comment thread and actually cropping with `args.resolution` return a square image while resizing will keep the aspect ratio. Maybe the mediapipe has a different behavior?

Reply:
here is the original code:
https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py#L846

this resizes to `resolution`, and then crops, also to `resolution`. Thus effectively, the crop doesnt do anything. The crop resolution should be < resolution for something to happen. This we think is a bug in the original code

To get around that we also add "crop_resolution". The idea being, first resize to "resolution" and then center crop or random crop to "crop_resolution". In the command line we keep both same, so that we have same behaviour as original diffuser code, where resolution and crop_resolution are same.

So how do you want this code to be like?
1. Go back to diffuser code, with one argument "resolution", in that case I think we should just remove the crop, because it does nothing.
2. Keep both, such that there is a possibility of random crop or center crop.

Reply:
The behavior is actually a bit different. [The doc of resize](https://pytorch.org/vision/main/generated/torchvision.transforms.Resize.html) says that "If size is an int, smaller edge of the image will be matched to this number. i.e, if height > width, then image will be rescaled to (size * height / width, size)." So if images are not square, only the smaller edge will be equal to `resolution`.

I would tend to stick to Diffusers code, but if having another argument for the crop resolution is better for you, then let's keep it.

Reply:
Hmm i see what you mean. However we have difficulties retesting it now that the dataset is no longer available. I think once we figure out a new dataset in the next release, I can come back to this and clean up reso/crop_reso

Reply:
Sounds good, let's do that